### PR TITLE
API Creation and Internship page

### PR DESCRIPTION
**Pull request by Ememobong28**
1.) Used my internships database to create an API.
2.) Designed and Created my Internship Page.
3.) Fetched the internship data from API and displayed it.
4.) Implemented Search feature for internships.
5.) When an internship is clicked it, you are navigated to the company career site for now, later on it will be modified.

Attached to this is video recordings of my API data and my internship page
**Internships Page**
[Internships Page.zip](https://github.com/Ememobong28/Capstone-Project-TechHive/files/11979643/Internships.Page.zip)

**API**
[API.zip](https://github.com/Ememobong28/Capstone-Project-TechHive/files/11979598/API.zip)

### CODE REVIEW

File: techhive/seeders/internships.json [hunk 1]
```diff
@@ -18,11 +18,69 @@
...
+        "company": "Meta",
+        "category": "Freshman Friendly, Sophomore Friendly"
+      },
+      {
+        "title": "Amazon Software Development Engineer Internship",
+        "picture": "/Amazon-Logo.jpg",
+        "link": "https://www.amazon.jobs/en/teams/internships-for-students",
+        "description": "The Amazon Software Development Engineer Internship offers a challenging and rewarding experience for undergraduate students interested in software development. As an intern, you will work on projects that have a direct impact on Amazon's customers, collaborate with talented engineers, and learn from experienced mentors. Join Amazon and be part of a team that is revolutionizing e-commerce and technology.",
+        "company": "Amazon",
+        "category": "Sophomore Friendly, Junior Friendly"
```
Comment (nicolec2626) [inline techhive/seeders/internships.json:49]: just an idea for future, maybe we can make this an array so it's easier to do multi-select filter in the future `"category": ["Sophomore Friendly", "Junior Friendly"]`

**Comment by nicolec2626** [comment]
This PR is pretty large, which makes it more difficult for reviewers to look through code / provide feedback and for you to review changes. (Note we also don't want PRs to be too small and granular, otherwise it's difficult to understand what's changing)

I would have probably broken this PR down into two or even three separate PRs, something like:
1) Create API and internship page
2) Fetch and display internship data on internship page
3) Enable internships to be searchable and clickable

**Comment by nicolec2626** [comment]
Also, please check out the **PR Content Template** I shared with you earlier, for more details on what to include in your PRs (description, milestone, test plan, etc) in the future

File: techhive/seeders/internships.json (hunk 1 above)
Comment (Ememobong28) [reply techhive/seeders/internships.json:49]: Okay sure, that makes sense

**Comment by Ememobong28** [comment]
> This PR is pretty large, which makes it more difficult for reviewers to look through code / provide feedback and for you to review changes. (Note we also don't want PRs to be too small and granular, otherwise it's difficult to understand what's changing)
> 
> I would have probably broken this PR down into two or even three separate PRs, something like:
> 
> 1. Create API and internship page
> 2. Fetch and display internship data on internship page
> 3. Enable internships to be searchable and clickable

Okay noted. Thank you

**Pull request merged**